src/Resolvers/Collections.ts

```diff
@@ -92,46 +92,38 @@ export class CollectionsResolver {
 
   @FieldResolver(type => [Collection])
   async relatedCollections(
-    @Arg("size", () => Int, { nullable: true }) size: number,
+    @Arg("size", () => Int, { defaultValue: 10 }) size: number,
     @Root() collection: Collection
   ): Promise<Collection[]> {
-    const query: { where: {}; take?: number } = { where: {} }
     const isArtistBased =
       collection.query.artist_ids && collection.query.artist_ids.length > 0
 
-    if (size) {
-      query.take = size + 1
-    }
-
     if (isArtistBased) {
-      query.where["query.artist_ids"] = {
-        $in: collection.query.artist_ids,
-      }
-
-      const artistResults = await this.repository.find(query)
-
-      const relatedArtistCollections = reject(artistResults, {
-        id: collection.id,
+      const artistResults = await this.repository.find({
+        where: {
+          slug: { $ne: collection.slug },
```

Comment:
Went for the  `$ne` query method here as @anandaroop suggested. This lets us avoiding having to use `reject`.